### PR TITLE
cosmos db

### PR DESCRIPTION
**Pull request by faddat**
This is a draft PR for cosmos db

-- basically it should just wait on cutting a release of cosmos-db

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]

**Comment by marbar3778** [comment]
@faddat could you resolve conflicts

File: Makefile [hunk 1]
```diff
@@ -46,18 +46,18 @@ lint:
 # bench is the basic tests that shouldn't crash an aws instance
 bench:
 	cd benchmarks && \
-		go test $(LDFLAGS) -tags cleveldb,rocksdb,boltdb,badgerdb -run=NOTEST -bench=Small . && \
-		go test $(LDFLAGS) -tags cleveldb,rocksdb,boltdb,badgerdb -run=NOTEST -bench=Medium . && \
```
Comment (robert-zaremba) [inline Makefile:50]: why removing `badgerdb`?

**Review by robert-zaremba — COMMENTED** [review]
Le'ts keep `badgerdb`

File: Makefile (hunk 1 above)
Comment (faddat) [reply Makefile:50]: it is no longer in cosmos-db because it's abandoned.

**Comment by faddat** [comment]
* https://github.com/cosmos/iavl/pull/535

It'd be a good idea to merge #535.  The reason that golangci-lint is passing in 3 seconds is that we don't check out the repository in ci and #535 fixes that.

**Pull request merged**